samples/wifi/connect/README.rst

```diff
@@ -0,0 +1,27 @@
+.. _wifi_sta_connect:
+
+Wi-Fi: Auto Connect
+####################
+
+Overview
+********
+
+Application demonstrating usage of STA connect APIs of Wi-Fi manager.
```

Comment:
Use a complete sentence, and fix some reST formatting (need a blank line before a list):
```
This sample demonstrates using the Wi-Fi manager STA connect APIs
to connect a board to an access point (AP), following these steps:

1. Get STA config to obtain the specified AP
2. Open STA
3. Scan the specified AP
4. Connect to the specified AP
```